builds create prompt ui and api

H-b8 opened the pull request:
Builds UI for `pages/prompt/new.js`.

MilcahHalili reviewed (changes requested):
This is a great start, @H-b8! Can you also build out the api route so that, on click of the submit button, a new prompt is created in the database?

H-b8 commented:
- Built off of the `api/prompt/create` you started and have a successful post to db. Test @ /prompt/new
- Refactored code in a few places to get rid of warnings in the console
- Style adjustments
- Refactored Prompt component to show author on index page
- Refactored `prompt/[id]` page

MilcahHalili commented on components/Prompt.js:
```diff
@@ -1,22 +1,35 @@
-      <ReactMarkdown source={prompt.content} />
+    <div onClick={() => handleClick()} className={Styles.prompt}>
+      <h2 className={Styles.prompth2}>{prompt.title}</h2>
+      <p className={Styles.instructorName}>By {authorName || 'Unknown Instrcutor'}</p>
```
Typo in `Instructor`!

MilcahHalili commented on pages/api/prompt/create.js:
```diff
@@ -5,14 +5,19 @@ export default function handler(req, res) {
       const prompt = await prisma.prompt.create({
         data: {
-          instructorId: 1,
+          authorId: user.id,
```
Thanks for removing the hard-coded value/making this dynamic!

MilcahHalili commented on pages/prompt/[id].js:
```diff
@@ -14,31 +13,18 @@ const QuillNoSSRWrapper = dynamic(import('react-quill'), {
-        .catch(err => console.error(err))
-    }
-  }
+    const promptData = await fetch(`../api/prompt/${router.query.id}`);
+    const res = await promptData.json();
+    setData(res);
```
Ooo this is so much cleaner, tysm!

MilcahHalili commented on pages/prompt/[id].js:
```diff
@@ -58,15 +44,16 @@ const Prompt = () => {
+    await e.preventDefault();
+    await createPost();
+    setPostContent('');
+    console.log('yes!');
```
Lol we can remove this console log now, thanks!

MilcahHalili commented on pages/prompt/[id].js:
```diff
@@ -1,5 +1,4 @@
 import React, { useEffect, useState } from "react"
-import ReactMarkdown from "react-markdown"
```
Nice!

MilcahHalili reviewed (changes requested):
Amazing! Excellent work, @H-b8. Just a couple tiny change requests, and then we'll be good to go.

H-b8 commented:
Good eye @MilcahHalili ! Console removed and typo fixed 😄

MilcahHalili reviewed (approved):
Well done!
